agents/appmotion_motor/appmotion_motors_agent.py

```diff
@@ -0,0 +1,1142 @@
+
+        **Task** - Get the position of the motor while it is currently in
+        motion. An estimate based on the calculated trajectory of the movement,
+        relative to the zero point.
+
+        Parameters:
+            motor (int): Determines which motor, either 1 or 2, 3 is for all
+                motors. (default 1)
+            pos_is_inches (bool): Whether to return positions in inches or not.
+                (default True)
+
+        Returns:
+            positions (list): The positions of the specified motors.
+        """
+
+        with self.lock.acquire_timeout(1, job=f"pos_while_moving_motor{params['motor']}") as acquired:
+            if not acquired:
+                self.log.warn(
+                    f'Could not pos_while_moving because lock held by {self.lock.job}')
+                return False, "Could not acquire lock"
+            if params['motor'] == 1:
+                i_positions = self.motor1.get_immediate_position(params['pos_is_inches'])
+            elif params['motor'] == 2:
+                i_positions = self.motor2.get_immediate_position(params['pos_is_inches'])
+            elif params['motor'] == 3:
+                i_positions = self.motor1.get_immediate_position(params['pos_is_inches'])
+                i_positions = self.motor2.get_immediate_position(params['pos_is_inches'])
+            else:
+                i_positions = None
+                print("Motor ID invalid argument")
+
+        return True, "Current motor positions: {}".format(i_positions)
+
+    @ocs_agent.param('motor', default=1, type=int)
+    @ocs_agent.param('verbose', default=True, type=bool)
+    def is_moving(self, session, params):
+        """is_moving(motor=1, verbose=True)
+
+        **Tasks** - Checks if motors are moving OR if limit switches are
+        tripped.
+
+        Parameters:
+            motor (int): Determines which motor, either 1 or 2, 3 is for all
+                motors. (default 1)
+            verbose (bool): Prints output from motor requests if True.
+                (default True)
+        """
+        with self.lock.acquire_timeout(1, job=f"is_moving_motor{params['motor']}") as acquired:
+            if not acquired:
+                self.log.warn(
+                    f"Could not check because lock held by {self.lock.job}")
+                return False
+            if params['motor'] == 1:
+                self.move_status1 = self.motor1.is_moving(params['verbose'])
+            elif params['motor'] == 2:
+                self.move_status2 = self.motor2.is_moving(params['verbose'])
+            elif params['motor'] == 3:
+                self.move_status = self.motor1.is_moving(params['verbose']) or self.motor2.is_moving(params['verbose'])
+            else:
+                print("Motor ID invalid argument")
+
+        if self.move_status:
+            return True, ("Motors are moving.", self.move_status)
+        else:
+            return True, ("Motors are not moving.", self.move_status)
```

Comment:
How is this task used? Currently it'll always return `True`, whether the motors are moving or not. I'd suggest putting the movement state into the `session.data` object so that the state can be inspected programmatically.

Also, how is the distinction between "moving" and "limit switches are tripped" made?

Reply:
Yeah, this task is confusing. I think the idea is to use the last element in the tuple of `("Motors are/n't moving.", self.move_status)` to determine movement status. We just don't want to send commands to the motors while they are moving. +1 on putting movement status into the `session.data` object

As far as the user is concerned, I don't think there is a way to tell the difference...which might be something worth doing. Hardware wise, there is a unique message sent when you trip the limit switches